jena-base/src/main/java/org/apache/jena/atlas/lib/EscapeStr.java

```diff
@@ -80,7 +80,9 @@ public static void stringEsc(AWriter out, String s, char quoteChar, boolean sing
                 // Multiline string.
                 if ( c == quoteChar ) {
                     quotesInARow++ ;
-                    if ( quotesInARow == 3 ) {
+                    if ( (quotesInARow == 3) || (!singleLineString && (i == len - 1)) ) {
```

Comment:
The check for `!singleLineString` here is not necessary, as it will always be true, given the earlier check a few lines above. This line should probably look like this:
```
                    if ( (quotesInARow == 3) || (i == len - 1) ) {
```